Search results page

miina opened the pull request:
This PR is based on #42.
Fixes #43.

Adds search results page with dynamic data using amp live list.

Todo:
- Styling might need fixes;

![screen shot 2018-04-18 at 8 33 34 pm](https://user-images.githubusercontent.com/3294597/38948300-d5168592-4347-11e8-8179-36d24b68feed.png)

miina committed:
Start search template.

miina committed:
Fix phpcs. Rearragne templates.

miina commented:
@mehigh Finished the template for now, the search CSS can be found from assets/css/search.css, likely that there is something that needs fixing. Also, there is no search button, should we create this (cc @kienstra)? Or how should the search work?

miina committed:
Add missing template.

miina committed:
Fix phpcs.

kienstra commented:
**Proposed About Search Button**

Hi @postphotos,
Could you please help with @miina's question about the search button?

>Also, there is no search button, should we create this (cc @kienstra)? Or how should the search work?

The [design](https://www.figma.com/file/L14jTE7iNGOE7DH57Ro7ncyu/Travel-Theme?node-id=2%3A118) doesn't look to have a search button on the search results page:

<img width="870" alt="no-search-button" src="https://user-images.githubusercontent.com/4063887/38971568-6eed187a-4361-11e8-8d8b-7d4e7357cef9.png">

And neither doest the [AMP Start theme on Firebase](https://ampstart-iframes.firebaseapp.com/templates/travel/travel-results.amp). What do you think of this possible "search" button that I created?

<img width="1405" alt="possible-search-button" src="https://user-images.githubusercontent.com/4063887/38972081-2a270554-4364-11e8-8bf9-0d70985397f1.png">

The button is a few pixels too tall, though.

```html
<button class="ampstart-btn rounded bold">Find Adventures</button>
```

Styling (though this wouldn't be added inline in the actual PR:
```css
element.style {
    background: rgba(0,0,0,.2);
    margin-left: 25px;
}
```

kienstra commented:
This applies to Issue #29.

miina committed:
Add search button.

miina committed:
Merge remote-tracking branch 'origin/develop' into feature/29-search_results

miina committed:
Merge develop & resolve conflicts.

miina committed:
Adjust mobile view for search.php.

miina commented:
@kienstra This should also be ready for review now, added the search button, too. Note that on the mobile view the inputs and filters are removed and tried to make something similar to the [static theme's view](https://ampstart-iframes.firebaseapp.com/templates/travel/travel-results.amp).
One more note -- currently there is no design / template for no results page, added just a text. Could this stay like that? Or should we create something better?

@mehigh Would be great if you could also review the template, specifically for HTML / CSS adjustments.

miina commented:
Hey @kienstra, I think that this should be ready for review as well. Added a button for searching, let me know if that seems to look fine.

I'll ask @mehigh separately to take a look from HTML / CSS perspective.

kienstra commented on assets/css/search.css:
```diff
@@ -2,7 +2,18 @@
 
 /*! normalize.css v5.0.0 | MIT License | github.com/necolas/normalize.css */html{font-family:sans-serif;line-height:1.15;-ms-text-size-adjust:100%;-webkit-text-size-adjust:100%}body{margin:0}article,aside,footer,header,nav,section{display:block}h1{font-size:2em;margin:.67em 0}figcaption,figure,main{display:block}figure{margin:1em 40px}hr{box-sizing:content-box;height:0;overflow:visible}pre{font-f ...
```
It looks like this is stylesheet is appearing correctly in the `<stye amp-custom>`. But there look to be some styling issues:

<img width="1399" alt="search-php-styling" src="https://user-images.githubusercontent.com/4063887/39014850-137cd4b4-43e1-11e8-8740-2647c909d24e.png">

mehigh replied:
I've fixed this by removing the inline left float and by adding the flex wrapper @miina.

kienstra commented on templates/entry/slim.php:
```diff
@@ -0,0 +1,22 @@
+		<span class="travel-results-result-text"><?php echo get_the_title(); ?></span>
+		<span class="travel-results-result-subtext h3">&bull;</span>
+		<span class="travel-results-result-subtext h3">$&nbsp;</span>
+		<span class="bold"><?php echo esc_html( get_post_meta( get_the_ID(), 'amp_travel_price', true ) ); ?></span>
```
What do you think about outputting the 3 lines above only if the `'adventure'` has a price?

kienstra replied:
Nice use of [entry/meta/price.php](https://github.com/xwp/travel/pull/45/commits/6e69f19c8e962a92c3777c67a7ea5c77ad03f8df#diff-a044d91ea4383c8b07d6cb9d97b277fcR1) for this.

kienstra commented on templates/entry/slim.php:
```diff
@@ -0,0 +1,22 @@
+		<span class="bold"><?php echo esc_html( get_post_meta( get_the_ID(), 'amp_travel_price', true ) ); ?></span>
+	</div>
+	<?php get_template_part( 'templates/entry/meta/slim-meta' ); ?>
+</div>
```
It looks like the `$` and price sometimes appear on a different line:

<img width="758" alt="styling-product" src="https://user-images.githubusercontent.com/4063887/39015823-1c3c7534-43e4-11e8-9abe-bed4b068f998.png">

The [Firebase demo](https://ampstart-iframes.firebaseapp.com/templates/travel/travel-results.amp) doesn't look to have this issue with the `$`. But it is possible for the `*` and title to be on the same line, like for "Bake Traditional Sweet Bread."

In my opinion, the most important thing would be to not have the `$` and price on separate lines, and the bullet appearing on a separate line isn't an issue. But I'm happy to defer to @postphotos for this.

kienstra commented on templates/entry/meta/slim-meta.php:
```diff
@@ -0,0 +1,22 @@
+	<span class="nowrap slim-location">
+		<span class="travel-icon travel-img-icon-map-pin-outline-gray"></span>
+		<?php echo esc_html( $location ); ?>
+	</span>
```
What do you think about outputting these 4 lines above only if there is a a location for the `'adventure'`?

kienstra commented on templates/entry/meta/slim-meta.php:
```diff
@@ -0,0 +1,22 @@
+<div class="h4 line-height-2">
+	<?php get_template_part( 'templates/entry/meta/slim-rating' ); ?>
+	<?php /* translators: %d: Number of reviews. */ ?>
+	<span class="travel-results-result-subtext mr1"><?php printf( esc_html__( '%d Reviews', 'travel' ), esc_html( $reviews->approved ) ); ?></span>
```
How about:
```php
esc_html( _n( '%d Review', '%d Reviews', $reviews->approved, 'travel' ) )
```
in place of:
```php
esc_html__( '%d Reviews', 'travel' )
```

Though it's not part of this PR's diff, could you please make a similar change to [functions.php](https://github.com/xwp/travel/blob/1a4375af8bfb3feeffcf5a5aeee5ae1465901361/includes/functions.php#L126)?

kienstra replied:
Thanks for making these changes!

kienstra commented on templates/navbar-search.php:
```diff
@@ -3,69 +3,70 @@
+						<div class="travel-results-navbar-subheading h6 caps line-height-4">
+							<?php echo ! empty( $_GET['s'] ) ? esc_attr( $_GET['s'] ) . '•' : ''; ?>
+							<?php echo ! empty( $_GET['start'] ) ? sprintf( 'From %s', esc_attr( $_GET['start'] ) ) : ''; ?>
+							<?php echo ! empty( $_GET['end'] ) ? sprintf( 'Until %s', esc_attr( $_GET['end'] ) ) : ''; ?>
```
Could you please have these 3 lines above use `esc_html()` instead of `esc_attr()`? The result is the same for the normal values, though.

kienstra commented on search.php:
```diff
@@ -1,55 +1,43 @@
-        <!--/ Results -->
-    </div>
+		<!-- Results -->
+		<section class="travel-results pb1 md-pt1">
```
It sounds like the `taxonomy.php` or `taxonomy-*.php` template will be mainly the same as this `search.php`. 

So it might be good to extract this `section.travel-results` into a separate template. But we can do that in #37.

miina replied:
Thanks for noting that, will add it with #37.

kienstra commented on templates/navbar-search.php:
```diff
@@ -3,69 +3,70 @@
+								<?php esc_html_e( 'Return', 'travel' ); ?>
 							</div>
 						</label>
+						<a href="<?php echo esc_url( site_url() ); ?>?s=<?php echo esc_attr( $_GET['s'] ); ?>" [href]="'<?php echo esc_url( site_url() ); ?>?s=' + fields_query_live + '&start=' + (fields_start || '') + '&end=' + (fields_end || '')" class="search-travel-btn travel-input travel-input-dark rounded ampstart-btn white bold">
```
It looks like entering date value(s) removes the text from the query. For example, in the screenshot below, the query doesn't include the text:
```
https://local.them/?s=&start=2018-04-02&end=2018-04-27
```

<img width="883" alt="date-values-query" src="https://user-images.githubusercontent.com/4063887/39017800-dabf2074-43e9-11e8-9371-b7e7b22ead3b.png">

miina replied:
Thanks for catching that!

kienstra reviewed (changes requested):
**Looks Good**
*Some Points Including Styling*

Hi @miina,
This looks good. Like you mentioned, it'd be great to have @mehigh's help with some styling issues mentioned below.

kienstra commented:
**Question About "No Results"**

Hi @postphotos,
Could you please help with @miina's question:
>One more note -- currently there is no design / template for no results page, added just a text. Could this stay like that? Or should we create something better?

Here's how that look at the moment:

<img width="885" alt="no-results" src="https://user-images.githubusercontent.com/4063887/39018344-603f9732-43eb-11e8-867f-9374eabd28e7.png">

miina committed:
PHP-related fixes from review.

miina commented:
Hey @kienstra, thanks again for reviewing, the issues that are not related to styling should be fixed now.

kienstra commented:
**Thanks, PR Approved Other Than Styling**

 Hi @miina,
Thanks for addressing those points. This is approved, other than the styling points [here](https://github.com/xwp/travel/pull/45#discussion_r182865234) and [here](https://github.com/xwp/travel/pull/45#discussion_r182871049).

mehigh committed:
29 - Added flex wrapper, removed left float

mehigh commented:
I've identified the reason the images look squashed. In the demo all of the images had a width of 2 and a height of 1, which corresponded to a 2:1 format image. Since the images in your screenshot were not 2:1 that didn't work. I'm switching this to use the correct aspect ratio by reading the width & height from the thumbnail meta.

mehigh committed:
29 - Add correct, dynamic image height & width attributes

mehigh commented:
The reason the "$" wasn't appearing alone in the demo theme https://www.ampstart.com/templates/travel/travel-results.amp was the fact that after the span containing the $ there was no whitespace or newline. The fix for this was simply removing the whitespace altogether. There was correctly a non-breakable space in there, it only needed to be alone in order for the effect to work correctly.

mehigh committed:
29 - Remove undeeded whitespace

mehigh commented:
@kienstra this should be good to go now, as i've handled all 3 styling issues reported - price wrapping on a separate line than the $, the grid and the image formats.

mehigh reviewed (approved):
All looks good to me.

kienstra committed:
Add new image width to account for mobile view.

When the viewport is about 640px,
this image shows at a width of about 600px.
Before, this used a 'thumbnail' size,
which is 150x150 by default.
So create and use this bigger size
for the search.php page,
and later the taxonomy.php page.

kienstra commented:
**Thanks!**

Hi @mehigh,
Thanks a lot for fixing the display of the images, adding a flex wrapper, and making the `$` always appear next to the price. This looks good.

kienstra commented:
**New Image Size To Account For 640px Display**

If it's alright, I [added a new image size](https://github.com/xwp/travel/pull/45/commits/997e836aae8db3739097408cd9177ead392fd09c) of `600x300`. When the viewport is 640px, the image is about 600px:

<img width="635" alt="image-roughly-600-300" src="https://user-images.githubusercontent.com/4063887/39099624-6e5ef614-4643-11e8-9f36-33b9bcf7933c.png">

Before, the images had a lower resolution at that size. This keeps the expected 2-1 ratio.
